### PR TITLE
feature/protobuf

### PR DESCRIPTION
**Pull request by skyitachi**
fix: heartbeat

almost complete

**What this PR does**:

**Which issue(s) this PR fixes**:

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

1. support dubbogo protobuf, compatible with dubbo protobuf

```

### CODE REVIEW

File: protocol/dubbo/codec.go [hunk 1]
```diff
@@ -1,40 +1,38 @@
-/*
```
Comment (zouyx) [inline protocol/dubbo/codec.go]: why remove this license?

File: protocol/dubbo/const.go [hunk 2]
```diff
@@ -0,0 +1,239 @@
+package dubbo
+
+import (
+	"github.com/pkg/errors"
+	"reflect"
+	"regexp"
+)
+
+/*
```
Comment (zouyx) [inline protocol/dubbo/const.go]: license should at the top of this file

File: protocol/dubbo/hessian.go [hunk 3]
```diff
@@ -0,0 +1,490 @@
+package dubbo
```
Comment (zouyx) [inline protocol/dubbo/hessian.go]: should add license

File: protocol/dubbo/package.go [hunk 4]
```diff
@@ -0,0 +1,175 @@
+package dubbo
```
Comment (zouyx) [inline protocol/dubbo/package.go]: add license

File: protocol/dubbo/proto.go [hunk 5]
```diff
@@ -0,0 +1,374 @@
+package dubbo
```
Comment (zouyx) [inline protocol/dubbo/proto.go]: add license

File: protocol/dubbo/proto/payload.proto [hunk 6]
```diff
@@ -0,0 +1,61 @@
+syntax = "proto3";
```
Comment (zouyx) [inline protocol/dubbo/proto/payload.proto]: license

File: protocol/dubbo/request.go [hunk 7]
```diff
@@ -0,0 +1,23 @@
+package dubbo
```
Comment (zouyx) [inline protocol/dubbo/request.go]: license

File: protocol/dubbo/response.go [hunk 8]
```diff
@@ -0,0 +1,30 @@
+package dubbo
```
Comment (zouyx) [inline protocol/dubbo/response.go]: license

File: protocol/dubbo/serialize.go [hunk 9]
```diff
@@ -0,0 +1,6 @@
+package dubbo
```
Comment (zouyx) [inline protocol/dubbo/serialize.go]: license

**Comment by skyitachi** [comment]
I should base develop branch and fix license, so I will close it and start a new one

**Pull request closed without merging**